crates/fluvio-smartengine/src/smartmodule/mod.rs

```diff
@@ -233,6 +238,45 @@ impl SmartModuleWithEngine {
         let aggregate = SmartModuleAggregate::new(self, params, accumulator, version)?;
         Ok(aggregate)
     }
+
+    fn create_smartmodule(
+        &self,
+        params: SmartModuleExtraParams,
+        version: i16,
+        context: &SmartModuleContextData,
+    ) -> Result<Box<dyn SmartModuleInstance>> {
+        if let Ok(filter) = self.create_filter(params.clone(), version) {
+            return Ok(Box::new(filter));
+        }
```

Comment:
what about error cases? It is possible that create_filter could fail for other reasons.  Should distinguish between true error and not matching fn signature

Reply:
I create a internal error type: that represent the two types of error:
1) Not matching signature
2) Failed to instantiate


For the first one it will try to continue with the next smartmodule type. With the second one it will return that error so it doesn't try the next smartmodule types.